Add CompletedAndNotFinished in enum YoTennis.Models.MatchState

yuriyoleynikov opened the pull request:
issue #27 partial 1

yuriyoleynikov committed:
Add CompletedAndNotFinished in enum YoTennis.Models.MatchState

Igorbek commented:
Also need to add:
- an event that corresponds to that action (finish incomplete match), maybe `IncompleteMatchFinished`
- a method in the `GameHandler` that handles that action
- a method in the `GameController` that accepts the action from the user

Igorbek commented:
For sharing functionality, you need to create a link like:

`/match/shared/{matchId}?d={datetime}&h={hash}`

where:
- `{matchId}` - guid of the match
- `{datetime}` - is `DateTime.UtcNow.ToBinary()`
- `{hash}` - is a cryptographic hash (for example SHA256 or SHA512) of (`{matchId}, {datetime}, {userPassword}`) encoded in base-64 string (`Convert.ToBase64`)

yuriyoleynikov committed:
-Add CompletedAndNotFinished button in Views/Match
-Fix enum MatchState
-Add StopGameEvent class
-Fix Index method in MatchController class

yuriyoleynikov committed:
Rename to Stop

yuriyoleynikov committed:
-Fix GameHandler class (add cheks)
-Fix tests
